source/common/http/conn_manager_impl.cc

```diff
@@ -2269,19 +2272,25 @@ bool ConnectionManagerImpl::shouldDeferRequestProxyingToNextIoCycle() {
 }
 
 void ConnectionManagerImpl::onDeferredRequestProcessing() {
+  if (streams_.empty()) {
+    return;
+  }
   requests_during_dispatch_count_ = 1; // 1 stream is always let through
   // Streams are inserted at the head of the list. As such process deferred
-  // streams at the back of the list first.
-  for (auto reverse_iter = streams_.rbegin(); reverse_iter != streams_.rend();) {
-    auto& stream_ptr = *reverse_iter;
-    // Move the iterator to the next item in case the `onDeferredRequestProcessing` call removes the
-    // stream from the list.
-    ++reverse_iter;
-    bool was_deferred = stream_ptr->onDeferredRequestProcessing();
+  // streams in the reverse order.
+  auto reverse_iter = std::prev(streams_.end());
+  bool at_first_element = reverse_iter == streams_.begin();
```

Comment:
nit: Since `at_first_element` is immediately assigned `reverse_iter == streams_.begin()` in the body of the `do` block, how about omitting this here. Either just `bool at_first_element`, or perhaps `bool at_first_element = false`. This avoid duplicating the `reverse_iter == streams_.begin()` on consecutive lines.

Reply:
Done